### PR TITLE
fix axes and docs

### LINKED ISSUE #3250: aruco: conflicting documentation in estimatePoseSingleMarkers
- OpenCV => 4.5.5

Documentation says:

> The marker corrdinate [sic] system is centered on the middle of the marker, with the Z axis perpendicular to the marker plane.

This agrees with the implementation.

But it also says:

> The coordinates of the four corners of the marker in its own coordinate system are: (0, 0, 0), (markerLength, 0, 0), (markerLength, markerLength, 0), (0, markerLength, 0)

This, to my mind, disagrees with the previous statement. (0,0,0) is the origin, which _is_ the center of the marker, not any of its corners. Perhaps "its own coordinate system" refers to a different coordinate system... which it should not.

I would suggest the following:

> The coordinates of the four corners of the marker in its own coordinate system are: (-markerLength/2, -markerLength/2, 0), (markerLength/2, -markerLength/2, 0), (markerLength/2, markerLength/2, 0), (-markerLength/2, markerLength/2, 0)

Or alternatively, to _show_ with a diagram where and in what order the corners are.

I am doubting myself here, because [recently someone reported that their code puts an axis triad at the **corner** of the marker](https://forum.opencv.org/t/origin-not-in-center-of-aruco-marker/8649)... and I don't see why their code would do that (the python equivalent puts it in the center), but it would agree with the *second* sentence from the documentation.

### PR DESCRIPTION
**Pull request by AleksandrPanov**
fixes #3250
update docs and tutorials for Aruco estimatePoseSingleMarkers()

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/aruco/tutorials/aruco_detection/aruco_detection.markdown [hunk 1]
```diff
@@ -286,7 +286,7 @@ translation vectors of the estimated poses will be in the same unit
 - The output parameters `rvecs` and `tvecs` are the rotation and translation vectors respectively, for each of the markers
 in `markerCorners`.
 
-The marker coordinate system that is assumed by this function is placed at the center of the marker
+The marker coordinate system that is assumed by this function is placed in the top left corner of the marker
```
Comment (asmorkalov) [inline modules/aruco/tutorials/aruco_detection/aruco_detection.markdown:289]: The same for documentation here.

**Comment by asmorkalov** [comment]
```
In file included from /build/precommit-contrib_linux64/3.4/opencv_contrib/modules/aruco/src/aruco.cpp:40:0:
/build/precommit-contrib_linux64/3.4/opencv_contrib/modules/aruco/include/opencv2/aruco.hpp: In constructor 'cv::aruco::EstimateParameters::EstimateParameters()':
/build/precommit-contrib_linux64/3.4/opencv_contrib/modules/aruco/include/opencv2/aruco.hpp:235:19: error: 'PatternPos' is not a class or namespace
         pattern = PatternPos::CCW_center;
                   ^
/build/precommit-contrib_linux64/3.4/opencv_contrib/modules/aruco/src/aruco.cpp: In function 'void cv::aruco::_getSingleMarkerObjectPoints(float, cv::OutputArray, cv::aruco::EstimateParameters)':
/build/precommit-contrib_linux64/3.4/opencv_contrib/modules/aruco/src/aruco.cpp:823:59: error: 'cv::aruco::EstimateParameters::PatternPos' is not a class or namespace
     if (estimateParameters.pattern == EstimateParameters::PatternPos::CW_top_left_corner) {
                                                           ^
/build/precommit-contrib_linux64/3.4/opencv_contrib/modules/aruco/src/aruco.cpp:829:64: error: 'cv::aruco::EstimateParameters::PatternPos' is not a class or namespace
     else if (estimateParameters.pattern == EstimateParameters::PatternPos::CCW_center) {
                                                                ^
make[2]: *** [modules/aruco/CMakeFiles/opencv_aruco.dir/src/aruco.cpp.o] Error 1
```

File: modules/aruco/include/opencv2/aruco.hpp [hunk 2]
```diff
@@ -220,6 +221,33 @@ CV_EXPORTS_W void detectMarkers(InputArray image, const Ptr<Dictionary> &diction
...
+        * The coordinates of the four corners of the marker in its own coordinate system are:
+        * (-markerLength/2, markerLength/2, 0), (markerLength/2, markerLength/2, 0),
+        * (markerLength/2, -markerLength/2, 0), (-markerLength/2, -markerLength/2, 0)
+        */
+    CCW_center,
+    /** @brief The marker coordinate system is centered on the top-left corner of the marker.
+        * The coordinates of the four corners of the marker in its own coordinate system are:
+        * (0, 0, 0), (markerLength, 0, 0),
+        * (markerLength, markerLength, 0), (0, markerLength, 0)
+        */
```
Comment (asmorkalov) [inline modules/aruco/include/opencv2/aruco.hpp:235]: I think that CCW and CW meaning should be covered too.

File: modules/aruco/src/aruco.cpp [hunk 3]
```diff
@@ -810,19 +810,30 @@ static void _identifyCandidates(InputArray _image, vector< vector< vector< Point
 
 
 /**
-  * @brief Return object points for the system centered in a single marker, given the marker length
+  * @brief Return object points for the system centered in a top left corner of single marker, given the marker length
```
Comment (asmorkalov) [inline modules/aruco/src/aruco.cpp:813]: Need to cover all function parameters. Center coordinate system is default behaviour, but not the only one.

File: modules/aruco/tutorials/aruco_detection/aruco_detection.markdown (hunk 1 above)
Comment (AleksandrPanov) [reply modules/aruco/tutorials/aruco_detection/aruco_detection.markdown:289]: fixed

File: modules/aruco/include/opencv2/aruco.hpp (hunk 2 above)
Comment (AleksandrPanov) [reply modules/aruco/include/opencv2/aruco.hpp:248]: fixed

File: modules/aruco/src/aruco.cpp (hunk 3 above)
Comment (AleksandrPanov) [reply modules/aruco/src/aruco.cpp]: fixed

**Review by asmorkalov — APPROVED** [review]

**Comment by asmorkalov** [comment]
@alalek The patch ready for merge.

File: modules/aruco/src/aruco.cpp [hunk 4]
```diff
@@ -810,19 +810,31 @@ static void _identifyCandidates(InputArray _image, vector< vector< vector< Point
 
 
 /**
-  * @brief Return object points for the system centered in a single marker, given the marker length
+  * @brief Return object points for the system centered in a middle (by default) or in a top left corner of single
+  * marker, given the marker length
   */
-static void _getSingleMarkerObjectPoints(float markerLength, OutputArray _objPoints) {
+static void _getSingleMarkerObjectPoints(float markerLength, OutputArray _objPoints,
+                                         EstimateParameters estimateParameters) {
```
Comment (alalek) [inline modules/aruco/src/aruco.cpp:817]: const reference.

File: modules/aruco/src/aruco.cpp [hunk 5]
```diff
@@ -1221,17 +1233,17 @@ class SinglePoseEstimationParallel : public ParallelLoopBody {
     public:
     SinglePoseEstimationParallel(Mat& _markerObjPoints, InputArrayOfArrays _corners,
                                  InputArray _cameraMatrix, InputArray _distCoeffs,
-                                 Mat& _rvecs, Mat& _tvecs)
+                                 Mat& _rvecs, Mat& _tvecs, EstimateParameters _estimateParameters)
```
Comment (alalek) [inline modules/aruco/src/aruco.cpp:1236]: const reference

File: modules/aruco/src/aruco.cpp [hunk 6]
```diff
@@ -1242,21 +1254,20 @@ class SinglePoseEstimationParallel : public ParallelLoopBody {
     InputArrayOfArrays corners;
     InputArray cameraMatrix, distCoeffs;
     Mat& rvecs, tvecs;
+    EstimateParameters estimateParameters;
```
Comment (alalek) [inline modules/aruco/src/aruco.cpp:1257]: const reference

File: modules/aruco/src/aruco.cpp [hunk 7]
```diff
@@ -1242,21 +1254,20 @@ class SinglePoseEstimationParallel : public ParallelLoopBody {
...
                                InputArray _cameraMatrix, InputArray _distCoeffs,
-                               OutputArray _rvecs, OutputArray _tvecs, OutputArray _objPoints) {
+                               OutputArray _rvecs, OutputArray _tvecs, OutputArray _objPoints,
+                               Ptr<EstimateParameters> estimateParameters) {
 
     CV_Assert(markerLength > 0);
 
     Mat markerObjPoints;
-    _getSingleMarkerObjectPoints(markerLength, markerObjPoints);
+    _getSingleMarkerObjectPoints(markerLength, markerObjPoints, *estimateParameters);
```
Comment (alalek) [inline modules/aruco/src/aruco.cpp:1270]: unchecked pointer unreference.

File: modules/aruco/include/opencv2/aruco.hpp [hunk 8]
```diff
@@ -219,7 +220,55 @@ struct CV_EXPORTS_W DetectorParameters {
...
+enum PatternPos {
+    /** @brief The marker coordinate system is centered on the middle of the marker.
+        * The coordinates of the four corners (CCW order) of the marker in its own coordinate system are:
+        * (-markerLength/2, markerLength/2, 0), (markerLength/2, markerLength/2, 0),
+        * (markerLength/2, -markerLength/2, 0), (-markerLength/2, -markerLength/2, 0).
+        *
+        * These pattern points define this coordinate system:
+        * ![Image with axes drawn](images/singlemarkersaxes.jpg)
+        */
+    CCW_center,
```
Comment (alalek) [inline modules/aruco/include/opencv2/aruco.hpp:239]: Missing prefix for enum and its values violates OpenCV coding style guide.

**Pull request merged**